specification/immunisation-fhir-api.yaml

```diff
@@ -2,40 +2,287 @@
+    - healthcare worker access
+    - patient access
+
+    The end user could be:
+    * a healthcare worker - in which case you must ensure they are authenticated and suitably authorised locally
+    * a patient - in which case you must ensure they are authenticated locally
+    * not present at all - for example as part of a back end process to check NHS numbers for data flowing from one system to another
+    
+    ## Errors
+    We use standard HTTP status codes to show whether an API request succeeded or not. They are usually in the range:
+
+    * 200 to 299 if it succeeded, including code 202 if it was accepted by an API that needs to wait for further action
+    * 400 to 499 if it failed because of a client error by your application
+    * 500 to 599 if it failed because of an error on our server
+    
+    Errors specific to each API are shown in the Endpoints section, under Response. See our [reference guide](https://digital.nhs.uk/developer/guides-and-documentation/reference-guide#http-status-codes) for more on errors.
+    
     ## Environments and testing
-    To be completed.
+    | Environment       | Base URL                                                               |
+    | ----------------- | ---------------------------------------------------------------------- |
+    | Sandbox           | https://sandbox.api.service.nhs.uk/immunisation-fhir-api    |
+    | Integration test  | https://int.api.service.nhs.uk/immunisation-fhir-api        |
+
     ## Onboarding
     To be completed.
   contact:
-    name: 'immunisation-fhir-api API Support'
-    url: 'https://digital.nhs.uk/developer/help-and-support'
+    name: immunisation-fhir-api API Support
+    url: https://digital.nhs.uk/developer/help-and-support
     email: api.management@nhs.net
 servers:
-  - url: 'https://sandbox.api.service.nhs.uk/immunisation-fhir-api'
+  - url: https://sandbox.api.service.nhs.uk/immunisation-fhir-api
     description: Sandbox environment.
-  - url: 'https://int.api.service.nhs.uk/immunisation-fhir-api'
+  - url: https://int.api.service.nhs.uk/immunisation-fhir-api
     description: Integration test environment.
-  - url: 'https://api.service.nhs.uk/immunisation-fhir-api'
+  - url: https://api.service.nhs.uk/immunisation-fhir-api
     description: Production environment.
-paths: {}
+paths:
+  /immunisation:
+    post:
+      summary: Create Immunisation record event
+      operationId: createImmunisationRecord
+      description: Creates a single immunisation record event.
+      parameters:
+        - $ref: '#/components/parameters/CorrelationID'
+        - $ref: '#/components/parameters/RequestID'
+      responses:
+        201:
+          description: Created
+          content:
+            application/fhir+json:
+              schema:
+                $ref: '#/components/schemas/OperationOutcome'
+              example:
+                $ref: components/OperationOutcome/201_create-success.json
+          headers:
+            Location:
```

Comment:
I don't think we will implement this. Let's remove this for now and only add things that we know we will implement.